src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -484,11 +526,33 @@ private Thread createUnexpectedExitHook(final ShutdownListener listener, final P
         // not want to go unnoticed.
         out.println("FATAL: VM exiting unexpectedly.");
```

Comment:
Moving this debug output back near the halt would be good.

It would also be useful to add information indicating that we're about to wait on the semaphore with `haltAfter`.